notebooks/07_TipMasks.ipynb

```diff
@@ -0,0 +1,172 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Pipetting with specific tips\n",
+    "Sometimes it is useful to specify exactly which tip to use for pipetting.\n",
+    "This can be to avoid cross-contaminations, but is __also needed if some tips are broken/deactivated__.\n",
+    "\n",
+    "To restrict pipetting operations to specific tips, one can pass the `tip` kwarg.\n",
+    "The `tip` kwarg can be used in three ways:\n",
+    "* Use `tip=1` to use only tip number one\n",
+    "* Alternatively `tip=robotools.Tip.T1` is the same as the above.\n",
+    "* Or you can pass an iterable such as a `set`, `tuple` or `list`. For example: `tip=[1, 3, 4]`\n",
+    "\n",
+    "The code examples below show this in action."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import robotools\n",
+    "\n",
+    "water = robotools.Trough(\"water\", 8, 1, min_volume=0, max_volume=100_000, initial_volumes=100_000)\n",
+    "target = robotools.Labware(\"target\", 4, 5, min_volume=0, max_volume=300)\n",
+    "\n",
+    "\n",
+    "with robotools.Worklist() as wl:\n",
+    "    wl.transfer(\n",
+    "        source=water,\n",
+    "        source_wells=robotools.get_trough_wells(10, water.wells),\n",
+    "        destination=target,\n",
+    "        # Only the 1st and 4th row\n",
+    "        destination_wells=[\n",
+    "            ['A01', 'A02', 'A03', 'A04', 'A05'],\n",
+    "            ['D01', 'D02', 'D03', 'D04', 'D05'],\n",
+    "        ],\n",
+    "        volumes=200,\n",
+    "        # Using tips 1 and 4\n",
+    "        tip=[1, 4],\n",
+    "        label=\"Transfer water with tips 1 and 4\",\n",
+    "    )"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The volumes are as expected: Water in the first and last row 👇"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "target\n",
+       "[[200. 200. 200. 200. 200.]\n",
+       " [  0.   0.   0.   0.   0.]\n",
+       " [  0.   0.   0.   0.   0.]\n",
+       " [200. 200. 200. 200. 200.]]"
+      ]
+     },
+     "execution_count": 2,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "target"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "If we look at the worklist `wl` we can see that each aspirate (A) and dispense (D) command had a tip mask of `9` which corresponds to the sum of the binary values for `Tip.T1` (1) and `Tip.T4` (8)."
```

Comment:
hier wäre vielleicht noch eine kleine Extrainfo gut, woher das mit den Werten kommt. Warum ist der Wert 8 für Tip 4, etc.?

Reply:
Ich habe es wie folgt geändert. Passt das so?

--------

Internally the EVOware identifies the tips in binary using the formula $2^{\text{tip number} - 1}$.
This way it can get away with storing one number instead of a sequence of numbers:

`Tip.T1` has the value $2^{1 - 1} = 1$ and `Tip.T4` has the value $2^{4 - 1} = 8$.

The tip mask for using both together is the sum: $1 + 8 = 9$.

If we look at the generated worklist `wl` we can see that each aspirate (A) and dispense (D) command had a tip mask of `9`.

Reply:
wunderbar. So kann man das auch eigenständig nachvollziehen :)